rmf_traffic_editor/gui/delete_dialog.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "delete_dialog.h"
+#include <algorithm>
+#include <QtWidgets>
+using std::vector;
+using std::string;
+
+
+DeleteDialog::DeleteDialog(
+  QWidget* parent,
+  Building& building,
+  int level_idx,
+  int selected_vertex_idx)
+: QDialog(parent)
+{
+
+
+  setWindowTitle("Warning");
+  QVBoxLayout* warning_message_vbox_layout = new QVBoxLayout;
+
+  // check if vertex is a lift cabin
+  std::vector<std::string> lift_levels;
+  auto level = building.levels[level_idx];
+  const auto v = level.vertices[selected_vertex_idx];
+  auto it = v.params.find("lift_cabin");
+  if ((it != v.params.end()))
+  {
+    auto lift = building.get_lift(v.lift_cabin());
+    for (const auto& level : building.levels)
+    {
+      if (level.elevation <= lift.highest_elevation &&
+        level.elevation >= lift.lowest_elevation)
+      {
+        lift_levels.push_back(level.name);
+      }
```

Comment:
I'm not quite sure this is the right way to check for the lift. Specifically, lifts can have levels that are skipped in-between, in which case this prints the wrong message.
To reproduce, try the following in the hotel world;

* Delete a lift vertex from one of the lifts
* Disable L2 visit in the lift as below:
![image](https://github.com/user-attachments/assets/37f79ec8-a19d-464b-b909-7bac4ef951ea)

* Add the lift waypoints
* Notice how L2 does _not_ get a lift waypoint
* Delete a waypoint from L3, note how it will say that there is a waypoint in L2 while that is not the case

Reply:
I made changes to use the `LevelDoorMap` instead.

But that also made me realised that when the door of that level is removed, the lift vertex at that level still persist.

Might want to include a cleanup action to remove that vertex.